rfd/0033-sni-alpn-teleport-proxy-routing.md

```diff
@@ -0,0 +1,221 @@
+                                                              │
+                                                              │
+                                                              │
+                                                              │
+                                            ┌─────────────────┴─────────────────┐
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            ▼                                   ▼
+                             ┌────────────────────────────┐    ┌─────────────────────────────────┐
+                             │SNI:                        │    │SNI:                             │
+                             │  proxy.example.teleport.com│    │ kube.proxy.example.teleport.com │
+                             │  default                   │    │                                 │
+                             └────────────────────────────┘    └─────────────────────────────────┘
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            │                                   │
+                                            │                                 ALPN:
+                                            │                         http/1.1, h2, default
+           ┌────────────────┬───────────────┴──┬───────────────┐                │
+           │                │                  │               │                │
+           │                │                  │               │                │
+         ALPN:              │                  │ALPN:          │                │
+    teleport-mysql          │            teleport-proxy-ssh    │                │
+           │                │                  │               │                │
+           │                │ALPN:             │             ALPN:              │
+           │           teleport-postgres       │     http/1.1, h2, default       │
+           │                │                  │               │                │
+           ▼                ▼                  ▼               ▼                ▼
+    ┌────────────┐  ┌───────────────┐  ┌───────────────┐  ┌─────────┐  ┌────────────────┐
+    │MySQL proxy │  │Postgres proxy │  │ssh proxy      │  │web proxy│  │kube proxy      │
```

Comment:
Could you also add ability to patch connection through to the auth server as well here as @klizhentas [asked before](https://github.com/gravitational/teleport/pull/7280#discussion_r650346329)?